### DISCUSSION TITLE
Obsidian File & Link Settings still active

### DISCUSSION THREAD

**Discussion by Joshs-Dev-Quest** [trganda/obsidian-attachment-management#70] [Q&A] [answered]
First thank you for the plugin I think it is a great idea.

Now I installed it and these are my settings:

Files & Links Settings from Obsidian itself
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/47bb648b-ab49-4d4b-a039-d84abe09f40b)

Attachement Managament (the plugin)
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/57257cbf-c464-4cbd-8239-0172eabf551a)

Also I restarted Obsidian once, I didn't restart my computer thought. 

I don't know maybe I set up something wrong,.
So the folder are generetated like that:
Attachements/${filename}/ 
But the copy pasted picture is not landing inside that folder, it's always going there where the Files & Links settings are set.
I tried to not copy the Obsidian settings too but there was the same problem.

Does somebody maybe know where I made the mistake?

Thanks have a nice day

**Comment by trganda** [comment]
I have copy your setting and tested on my pc, it's working.

If you notes was placed in folder `test` with name `Untitled`, after paste image file it should be placed in `test/Attachements/Untitled/xxx.png`

<img width="460" alt="image" src="https://github.com/trganda/obsidian-attachment-management/assets/62204882/efb738ed-a33f-4cac-a017-e114fdf9036e">


As your description, your pasted image file was placed in `test/Attachements `, not `test/Attachements/Untitled` right?

I'm not sure where is the problem, have you tried the deault setting of this plugin?

**Comment by Joshs-Dev-Quest** [comment]
Hey @trganda

Thank you for your fast reply.

Ok I tried it again with the default settings and then I played arround a little and now with this settings
Attachment path: ${notepath}/Attachments/${notename} 
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/2e5deabb-137b-467f-b177-43e38cb9b1f6)
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/b2a69abc-501c-47e1-bd26-bad44adc4b08)


it's like this
I can create a new note
The first time I paste an image in this note this happens
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/ff8d3e0d-d666-47f0-9129-facd7b555bd0)

The folder gets created, but the image is nowhere to be found, it don't get safed. 
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/d5aed6e7-d329-46a5-be4b-fd25b1aedb1f)

After this I can paste a second picture into the file:
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/e41b00cf-60b9-4aa1-a66b-1b99d8e88ba9)

And then the picture is in the wanted folder with the name -1 (the third picture would be -2 and so on)
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/8f9c809c-3a3a-4726-a15e-4f1bcf93b8ac)


So I opened a sandbox vault and there it worked.

Maybe this plugin is not working together with one of my previous installed one?
Or is it another setting of my vault maybe?

Anyway I will use your plugin and to the last move manually.
When I can help you somehow to erase that bug let me know, I will try.

I will send you my last try at the end

List of plugins:
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/7ed3cea4-69db-400b-8612-d6a232bf44d4)
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/cb9fb0f2-d0cd-4037-a2be-559b58fc85c7)
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/5f63b665-ca2c-43ba-a1e7-2aaa5abf33b7)
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/dc5ea27f-1329-410e-88da-fbec52546be9)

My wanted settings in the sandbox vault
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/40c3e0bb-72fa-417e-b63e-02e6a5d71a72)
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/78521fe5-5172-4735-8895-9059b690a664)


With the default settings, the renaming process is working with the -1. -2 ....
and when I set the img-${date} inside then the pasted image isn't going into the wanted folder but is named correctly
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/776b5e88-7245-43f9-864b-2dd545b0176c)
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/7ce58b48-2da2-45f3-898e-2780d6df7b9f)

**Reply by trganda** [reply, marked as answer]
Have you only enabled this plugin in sandbox right?

The pasted image has been renamed to .png, -1.png ... because the setting was empty, the content in the input box was only a place holder

<img width="786" alt="image" src="https://github.com/trganda/obsidian-attachment-management/assets/62204882/475883e1-c451-4d34-9a12-3b18c05221d9">

when you first time install this plugin, the default setting should as below

<img width="756" alt="image" src="https://github.com/trganda/obsidian-attachment-management/assets/62204882/6d744e8d-8bdc-4d14-9cd7-9679d1703802">

If you delete the content in the input box, the seting has been empyt, you can check the `data.json` file in your `<you vault folder>/.obsidian/plugins/attachment-management/data.json` (this file will not exists until you change the setting).

As you wanted setting (If only enabled this plugin), the `data.json` and `app.json` (obsidian setting file `<you vault folder>/.obsidian/app.json`) should be like this and it's work on my pc

```json
{
  "attachPath": {
    "attachmentRoot": "",
    "saveAttE": "obsFolder",
    "attachmentPath": "${notename}",
    "attachFormat": "IMG-${date}",
    "type": "GLOBAL"
  },
  "dateFormat": "YYYYMMDDHHmmssSSS",
  "excludeExtensionPattern": "pdf",
  "autoRenameAttachment": false,
  "excludedPaths": "",
  "excludePathsArray": [],
  "excludeSubpaths": false,
  "overridePath": {}
}
```

`app.json`

```json
{
  "alwaysUpdateLinks": true,
  "attachmentFolderPath": "./Attachments",
  "showUnsupportedFiles": true,
  "newLinkFormat": "absolute"
}
```

If this was still not working in your sandbox vault, I will try to find this bug with other plugins that you supplied.

**Reply by Joshs-Dev-Quest** [reply]
Yes in the sandbox vault I only downloaded this plugin and nothing else. There everything is working fine and there is no problem.
Sorry for the misunderstanding.


Only on my normal vault it's not working.
But I looked up app.json and in my normal vault the settings are these:
![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/0a0f98f4-fefc-4bf5-95b6-769403920a4e)

![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/5c69ed6f-756b-479e-b8c7-9a5f28f7e197)

![image](https://github.com/trganda/obsidian-attachment-management/assets/101974582/026efbc9-d372-45b9-b4a2-a2ec14a947e3)

**Comment by Joshs-Dev-Quest** [comment]
Hey @trganda Thank you a lot for your effort.
You encouraged me for still trying everything and not giving up.

But I am really sorry to tell you that every thing is fine now because I reinstalled (I think it's that) the plugin.

Sorry for wasting your time but I appreciate your help and will use your plugin

**Reply by trganda** [reply]
Happy to hear that. :)
